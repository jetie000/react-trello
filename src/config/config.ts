export const config = {
  API_URL: process.env.REACT_APP_API_URL,
  BURGER_BREAKPOINT: 576
}
