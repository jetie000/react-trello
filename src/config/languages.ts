export const languages = [
  {
    ERROR: "Ошибка",
    ERROR_ACCESS_TOKEN: "Ошибка токена доступа",
    INPUT_DATA: "Введите данные",
    LOADING: "Загрузка...",
    LOGIN: "Вход",
    REGISTER: "Регистрация",
    INVALID_LOGIN: "Вы ввели неправильный email или пароль",
    NO_ACCESS: "Нет доступа",
    SUCCESFULLY_ENTERED: "Вы успешно вошли",
    ENTER_EMAIL: "Введите e-mail",
    PASSWORD: "Пароль",
    ENTER_PASSWORD: "Введите пароль",
    LOG_IN: "Войти",
    ALREADY_EXISTS: "Пользователь с таким адресом эл. почты существует",
    SUCCESFULLY_REGISTERED:
      "Вы успешно зарегистрированы. Проверьте свой e-mail для подтверждения аккаунта",
    SUCCESS: "Успешно",
    SURNAME: "Фамилия",
    ENTER_SURNAME: "Введите фамилию",
    NAME: "Имя",
    ENTER_NAME: "Введите имя",
    REGISTER_: "Зарегистрироваться",
    ERROR_USER_DATA: "Ошибка в данных пользователя",
    USER_SUCCESSFULLY_DELETED: "Пользователь успешно удален",
    USER_NOT_FOUND: "Пользователь не найден",
    USER_SUCCESSFULLY_CHANGED: "Пользователь успешно изменен",
    SURE_DELETE_ACC: "Вы точно хотите удалить аккаунт? Вместе с этим вы удалите все его коллекции.",
    DELETE_ACCOUNT: "Удалить аккаунт",
    DELETING_ACCOUNT: "Удаление аккаунта",
    NEW_PASSWORD: "Новый пароль (По желанию)",
    ENTER_NEW_PASSWORD: "Введите новый пароль",
    CHANGE_DATA: "Изменить данные",
    DELETE_USER: "Удалить пользователя",
    MY_DATA: "Мои данные",
    LOGOUT: "Выйти",
    WRONG_PASSWORD: "Вы ввели неправильный пароль",
    DATA_SUCCESFULLY_CHANGED: "Данные успешно изменены",
    ERROR_IN_USER_DATA: "Ошибка в данных пользователя",
    SURE_DELETE_ACC_MY:
      "Вы точно хотите удалить аккаунт? Вместе с этим вы удалите все свои коллекции.",
    ENTER_NEW_SURNAME: "Введите новую фамилию",
    ENTER_NEW_NAME: "Введите новое имя",
    ENTER_NEW_EMAIL: "Введите новый email",
    ENTER_OLD_PASS: "Введите старый пароль",
    OLD_PASS: "Старый пароль (Обязательно)",
    NOTIFICATION: "Уведомление",
    ENTER_REQUEST: "Введите запрос",
    ERROR_REQUEST_BY_IDS: "Ошибка запроса по id пользователей",
    ERROR_REQUEST_SEARCH: "Ошибка запроса пользователей",
    SURNAME_AND_NAME: "Фамилия и имя",
    ENTER_SURNAME_NAME: "Введите фамилию и имя",
    MY_BOARDS: "Мои доски",
    DARK: "Темная",
    LIGHT: "Светлая",
    ADD_BOARD: "Добавить доску",
    CABINET: "Кабинет",
    MENU: "Меню",
    ERROR_404: "Ошибка 404 - Страница не найдена",
    ERROR_REQUEST: "Ошибка запроса",
    BOARD_ADDED: "Доска успешно добавлена",
    DESCRIPTION: "Описание (по желанию)",
    ENTER_DESCRIPTION: "Введите описание",
    BOARD_DELETED: "Доска успешно удалена",
    ERROR_BOARD_DELETING: "Ошибка в удалении доски",
    BOARD_LEFT: "Вы успешно покинули доску",
    ERROR_BOARD_LEAVING: "Ошибка в покидании доски",
    SURE_DELETE_BOARD:
      "Вы уверены, что хотите удалить эту доску? Вы потеряете все столбцы и задачи этой доски.",
    DELETE_BOARD: "Удалить доску",
    SURE_LEAVE_BOARD: "Вы уверены, что хотите покинуть эту доску?",
    LEAVE_BOARD: "Покинуть доску",
    CREATOR: "Создатель",
    BOARD_CHANGED: "Доска успешно изменена",
    CHANGE_BOARD: "Изменить доску",
    COLUMN_ADDED: "Столбец успешно добавлен",
    ADD_COLUMN: "Добавить столбец",
    COLUMN_CHANGED: "Столбец успешно изменен",
    CHANGE_COLUMN: "Изменить столбец",
    COLUMN_DELETED: "Столбец успешно удален",
    SURE_DELETE_COLUMN: "Вы уверены, что хотите удалить столбец? Вы потеряете все задачи столбца.",
    DELETE_COLUMN: "Удалить столбец",
    ERROR_CHANGING_TASK_COL: "Ошибка изменения столбца задачи",
    ERROR_CHANGING_COL_ORDER: "Ошибка изменения порядка столбцов",
    TASK_ADDED: "Задача успешно добавлена",
    ADD_TASK: "Добавить задачу",
    TASK_CHANGED: "Задача успешно изменена",
    CHANGE_TASK: "Изменить задачу",
    TASK_DELETED: "Задача успешно удалена",
    SURE_DELETE_TASK: "Вы уверены, что хотите удалить задачу?",
    DELETE_TASK: "Удалить задачу",
    MOVED: "Перемещено",
    CREATED: "Создано",
    USERS: "Польз.",
    ADD_USERS: "..."
  },
  {
    ERROR: "Error",
    ERROR_ACCESS_TOKEN: "Access token error",
    INPUT_DATA: "Input data",
    LOADING: "Loading...",
    LOGIN: "Log In",
    REGISTER: "Sign Up",
    INVALID_LOGIN: "You've entered invalid email or password",
    NO_ACCESS: "No access",
    SUCCESFULLY_ENTERED: "You've succesfully entered",
    ENTER_EMAIL: "Input e-mail",
    PASSWORD: "Password",
    ENTER_PASSWORD: "Input password",
    LOG_IN: "Log In",
    ALREADY_EXISTS: "User with that email already exists",
    SUCCESFULLY_REGISTERED: "You've succesfully registered. Check your e-mail for confirm letter",
    SUCCESS: "Success",
    SURNAME: "Surname",
    ENTER_SURNAME: "Input surname",
    NAME: "Name",
    ENTER_NAME: "Input name",
    REGISTER_: "Sign Up",
    ERROR_USER_DATA: "User data error",
    USER_SUCCESSFULLY_DELETED: "User successfully deleted",
    USER_NOT_FOUND: "User not found",
    USER_SUCCESSFULLY_CHANGED: "User successfully changed",
    SURE_DELETE_ACC:
      "Are you sure you want to delete an account? You will also delete all of his collections.",
    DELETE_ACCOUNT: "Delete an account",
    DELETING_ACCOUNT: "Account deleting",
    NEW_PASSWORD: "New password (Optional)",
    ENTER_NEW_PASSWORD: "Input new password",
    CHANGE_DATA: "Change data",
    DELETE_USER: "Delete user",
    MY_DATA: "My info",
    LOGOUT: "Log out",
    WRONG_PASSWORD: "You've entered wrong password",
    DATA_SUCCESFULLY_CHANGED: "Info succesfully changed",
    ERROR_IN_USER_DATA: "Error in user data",
    SURE_DELETE_ACC_MY:
      "Are you sure you want to delete an account? You will also delete all of your collections.",
    ENTER_NEW_SURNAME: "Input new surname",
    ENTER_NEW_NAME: "Input new name",
    ENTER_NEW_EMAIL: "Input new email",
    ENTER_OLD_PASS: "Input old password",
    OLD_PASS: "Old password (necessary)",
    NOTIFICATION: "Notification",
    ENTER_REQUEST: "Enter request",
    ERROR_REQUEST_BY_IDS: "Request by ids error",
    ERROR_REQUEST_SEARCH: "Request search error",
    SURNAME_AND_NAME: "Surname and name",
    ENTER_SURNAME_NAME: "Enter surname and name",
    MY_BOARDS: "My boards",
    DARK: "Dark",
    LIGHT: "Light",
    ADD_BOARD: "Add board",
    CABINET: "Cabinet",
    MENU: "Menu",
    ERROR_404: "Error 404 - Page not found",
    ERROR_REQUEST: "Request error",
    BOARD_ADDED: "Board succesfully added",
    DESCRIPTION: "Description (optional)",
    ENTER_DESCRIPTION: "Enter description",
    BOARD_DELETED: "Board deleted successfully",
    ERROR_BOARD_DELETING: "Error in board deleting",
    BOARD_LEFT: "Board left successfully",
    ERROR_BOARD_LEAVING: "Error in board leaving",
    SURE_DELETE_BOARD:
      "Are you sure you want delete this board? You will lost all your columns and tasks dedicated to this board.",
    DELETE_BOARD: "Delete board",
    SURE_LEAVE_BOARD: "Are you sure you want leave this board?",
    LEAVE_BOARD: "Leave board",
    CREATOR: "Creator",
    BOARD_CHANGED: "Board succesfully changed",
    CHANGE_BOARD: "Change board",
    COLUMN_ADDED: "Column added succesfully",
    ADD_COLUMN: "Add column",
    COLUMN_CHANGED: "Column changed succesfully",
    CHANGE_COLUMN: "Change column",
    COLUMN_DELETED: "Column deleted succesfully",
    SURE_DELETE_COLUMN:
      "Are you sure you want delete this column? You will lost all tasks dedicated to this column.",
    DELETE_COLUMN: "Delete column",
    ERROR_CHANGING_TASK_COL: "Error changing task column",
    ERROR_CHANGING_COL_ORDER: "Error changing column order",
    TASK_ADDED: "Task added succesfully",
    ADD_TASK: "Add task",
    TASK_CHANGED: "Task changed succesfully",
    CHANGE_TASK: "Change task",
    TASK_DELETED: "Task deleted succesfully",
    SURE_DELETE_TASK: "Are you sure you want delete this task?",
    DELETE_TASK: "Delete task",
    MOVED: "Moved",
    CREATED: "Created",
    USERS: "Users",
    ADD_USERS: "add users"
  }
]
