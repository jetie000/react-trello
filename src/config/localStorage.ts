export const localStorageConfig = {
  TOKEN_LOCALSTORAGE: "token",
  THEME_LOCALSTORAGE: "page_theme",
  LANGUAGE_LOCALSTORAGE: "page_language",
  USERID_LOCALSTORAGE: "user_id"
}
