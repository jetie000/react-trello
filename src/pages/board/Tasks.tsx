import { IColumn } from '@/types/column.interface';
import * as React from 'react';

function Tasks({ column }: { column: IColumn }) {

    const tasksSorted = React.useMemo(
        () => column.tasks && column.tasks.slice()
            .sort((t1, t2) => t2.moveDate.getMilliseconds() - t1.moveDate.getMilliseconds())
        , [column.tasks])

    return (
        <div className='rounded-2 d-flex flex-column'>
            {
                tasksSorted && tasksSorted.map(t =>
                    <div>
                        {t.name}
                    </div>
                )
            }
            <button className='btn btn-primary border rounded-2'>
                <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" fill="currentColor" className="bi bi-plus" viewBox="0 0 16 16">
                    <path d="M8 4a.5.5 0 0 1 .5.5v3h3a.5.5 0 0 1 0 1h-3v3a.5.5 0 0 1-1 0v-3h-3a.5.5 0 0 1 0-1h3v-3A.5.5 0 0 1 8 4" />
                </svg>
            </button>
        </div>
    );
}

export default Tasks;