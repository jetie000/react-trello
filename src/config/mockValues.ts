import { IBoard } from "@/types/board.interface"
import { IColumn } from "@/types/column.interface"
import { IUserResponse } from "@/types/user.interface"

export const mockColumn: IColumn = {
  id: 1,
  boardId: 1,
  name: "TODO",
  order: 1,
  tasks: [
    {
      id: 1,
      columnId: 1,
      name: "Task 1",
      creationDate: new Date(0),
      moveDate: new Date(0),
      description: "Description of Task 1",
      users: [
        { id: 1, fullName: "User 1", email: "user1@gmail.com", loginDate: new Date() },
        { id: 2, fullName: "User 2", email: "user2@gmail.com", loginDate: new Date() },
        { id: 3, fullName: "User 3", email: "user3@gmail.com", loginDate: new Date() }
      ],
      Column: {
        id: 1,
        boardId: 1,
        name: "TODO",
        order: 1
      }
    },
    {
      id: 2,
      columnId: 1,
      name: "Task 2",
      creationDate: new Date(),
      moveDate: new Date(),
      description: "Description of Task 2",
      users: [
        { id: 4, fullName: "User 4", email: "user4@gmail.com", loginDate: new Date() },
        { id: 5, fullName: "User 5", email: "user5@gmail.com", loginDate: new Date() }
      ],
      Column: {
        id: 1,
        boardId: 1,
        name: "TODO",
        order: 1
      }
    }
  ]
}

export const mockUsers = [
  {
    id: 1,
    email: "1@gmail.com",
    fullName: "AA AA",
    loginDate: new Date()
  },
  {
    id: 2,
    email: "2@gmail.com",
    fullName: "BB BB",
    loginDate: new Date()
  },
  {
    id: 3,
    email: "3@gmail.com",
    fullName: "CC CC",
    loginDate: new Date()
  }
]
export const mockReturnUsers = (users: IUserResponse[]) => ({
  isLoading: false,
  isError: false,
  refetch: jest.fn(),
  data: users
})
export const mockBoards: IBoard[] = [
  {
    id: 1,
    creatorId: 1,
    name: "Board1",
    columns: [
      mockColumn,
      {
        id: 2,
        boardId: 1,
        name: "PENDING",
        order: 2
      },
      {
        id: 3,
        boardId: 1,
        name: "DONE",
        order: 3
      }
    ]
  },
  {
    id: 2,
    creatorId: 1,
    name: "Board2"
  },
  {
    id: 3,
    creatorId: 1,
    name: "Board3"
  }
]

export const mockReturnMutation = (isLoading: boolean, isPressed: boolean) => ({
  isLoading: isLoading && isPressed,
  isError: false,
  isSuccess: !isLoading && isPressed,
  refetch: jest.fn(),
  reset: jest.fn()
})
