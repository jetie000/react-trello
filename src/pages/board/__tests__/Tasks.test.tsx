import * as reduxHooks from "react-redux"
import { fireEvent, render, screen } from "@testing-library/react"
import { IColumn } from "@/types/column.interface"
import Tasks, { TasksProps } from "../Tasks"
import { mockColumn } from "../../../config/mockValues"

jest.mock("@/store/store")
jest.mock("react-redux")
const mockUseSelector = jest.spyOn(reduxHooks, "useSelector")
const mockReturnSelectorValue = {
  language: 1,
  id: 1,
  token: "token"
}

const renderTasks = (params: TasksProps) => render(<Tasks {...params} />)

const setCurrentColumn = jest.fn()
const setDrugStartTask = jest.fn()
const setCurrentTask = jest.fn()

describe("Tasks", () => {
  beforeEach(() => {
    jest.restoreAllMocks()
    jest.resetAllMocks()
    mockUseSelector.mockReturnValue(mockReturnSelectorValue)
  })
  it("should render a list of given tasks", () => {
    const component = renderTasks({
      column: mockColumn,
      setCurrentTask,
      setDrugStartTask,
      setCurrentColumn
    })
    const taskElements = screen.getAllByTestId(/task[0-9+]/)
    expect(taskElements.length).toBe(2)
    expect(component).toMatchSnapshot()
  })
  it("should render only the add task button when column has no tasks", () => {
    const column = {
      id: 1,
      boardId: 1,
      name: "TODO",
      order: 1,
      tasks: []
    }
    const component = renderTasks({ column, setCurrentTask, setDrugStartTask, setCurrentColumn })

    const taskElements = screen.queryAllByTestId(/task[0-9+]/)
    expect(taskElements.length).toBe(0)

    const addButton = screen.getByTestId("show-add-task")
    expect(addButton).toBeInTheDocument()
    expect(component).toMatchSnapshot()
  })
  it("should set the current task as the drag start task when dragging a task", () => {
    renderTasks({ column: mockColumn, setCurrentTask, setDrugStartTask, setCurrentColumn })

    const taskElement = screen.getByText("Task 1")
    fireEvent.dragStart(taskElement)

    expect(setDrugStartTask).toHaveBeenCalledWith(mockColumn.tasks![0])
  })
  it('should display "add users" when task has no users', () => {
    const column: IColumn = {
      id: 1,
      name: "Column 1",
      boardId: 1,
      order: 1,
      tasks: [
        {
          id: 1,
          name: "Task 1",
          description: "Description of Task 1",
          users: [],
          creationDate: new Date(),
          moveDate: new Date(),
          Column: { id: 1, name: "Column 1", order: 1, boardId: 1 },
          columnId: 1
        }
      ]
    }
    const component = renderTasks({ column, setCurrentTask, setDrugStartTask, setCurrentColumn })

    const addUserElement = screen.getByText(/add users/i)
    expect(addUserElement).toBeInTheDocument()
    expect(component).toMatchSnapshot()
  })
})
