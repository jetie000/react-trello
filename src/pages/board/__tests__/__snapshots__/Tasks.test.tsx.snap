// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tasks should display "add users" when task has no users 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rounded-2 d-flex flex-column"
      >
        <div
          class="d-flex flex-column board-task cursor-pointer cursor-grab"
          data-bs-target="#changeTask"
          data-bs-toggle="modal"
          data-testid="task1"
          draggable="true"
        >
          <span
            class="fs-5"
          >
            Task 1
          </span>
          <span>
            Moved
            :
          </span>
          <span
            class="text-truncate"
          >
            now
          </span>
          <span>
            Created
            :
          </span>
          <span
            class="text-truncate"
          >
            now
          </span>
          <div
            class="d-flex"
          >
            <span>
              Users
              :
            </span>
            <div
              class="ms-auto d-flex gap-1"
            >
              add users
            </div>
          </div>
        </div>
        <hr
          class="mb-2 mt-2"
        />
        <button
          class="btn btn-primary border rounded-2"
          data-bs-target="#addTask"
          data-bs-toggle="modal"
          data-testid="show-add-task"
        >
          <svg
            class="bi bi-plus"
            fill="currentColor"
            height="24"
            viewBox="0 0 16 16"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 4a.5.5 0 0 1 .5.5v3h3a.5.5 0 0 1 0 1h-3v3a.5.5 0 0 1-1 0v-3h-3a.5.5 0 0 1 0-1h3v-3A.5.5 0 0 1 8 4"
            />
          </svg>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="rounded-2 d-flex flex-column"
    >
      <div
        class="d-flex flex-column board-task cursor-pointer cursor-grab"
        data-bs-target="#changeTask"
        data-bs-toggle="modal"
        data-testid="task1"
        draggable="true"
      >
        <span
          class="fs-5"
        >
          Task 1
        </span>
        <span>
          Moved
          :
        </span>
        <span
          class="text-truncate"
        >
          now
        </span>
        <span>
          Created
          :
        </span>
        <span
          class="text-truncate"
        >
          now
        </span>
        <div
          class="d-flex"
        >
          <span>
            Users
            :
          </span>
          <div
            class="ms-auto d-flex gap-1"
          >
            add users
          </div>
        </div>
      </div>
      <hr
        class="mb-2 mt-2"
      />
      <button
        class="btn btn-primary border rounded-2"
        data-bs-target="#addTask"
        data-bs-toggle="modal"
        data-testid="show-add-task"
      >
        <svg
          class="bi bi-plus"
          fill="currentColor"
          height="24"
          viewBox="0 0 16 16"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 4a.5.5 0 0 1 .5.5v3h3a.5.5 0 0 1 0 1h-3v3a.5.5 0 0 1-1 0v-3h-3a.5.5 0 0 1 0-1h3v-3A.5.5 0 0 1 8 4"
          />
        </svg>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Tasks should render a list of given tasks 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rounded-2 d-flex flex-column"
      >
        <div
          class="d-flex flex-column board-task cursor-pointer cursor-grab"
          data-bs-target="#changeTask"
          data-bs-toggle="modal"
          data-testid="task1"
          draggable="true"
        >
          <span
            class="fs-5"
          >
            Task 1
          </span>
          <span>
            Moved
            :
          </span>
          <span
            class="text-truncate"
          >
            54 years ago
          </span>
          <span>
            Created
            :
          </span>
          <span
            class="text-truncate"
          >
            54 years ago
          </span>
          <div
            class="d-flex"
          >
            <span>
              Users
              :
            </span>
            <div
              class="ms-auto d-flex gap-1"
            >
              <div
                class="d-flex board-users-avatar-sm bg-secondary text-light border rounded-circle align-items-center justify-content-center"
              >
                U
              </div>
              <div
                class="d-flex board-users-avatar-sm bg-secondary text-light border rounded-circle align-items-center justify-content-center"
              >
                U
              </div>
              <div
                class="d-flex board-users-avatar-sm bg-secondary text-light border rounded-circle align-items-center justify-content-center"
              >
                U
              </div>
            </div>
          </div>
        </div>
        <div
          class="d-flex flex-column board-task cursor-pointer cursor-grab"
          data-bs-target="#changeTask"
          data-bs-toggle="modal"
          data-testid="task2"
          draggable="true"
        >
          <span
            class="fs-5"
          >
            Task 2
          </span>
          <span>
            Moved
            :
          </span>
          <span
            class="text-truncate"
          >
            now
          </span>
          <span>
            Created
            :
          </span>
          <span
            class="text-truncate"
          >
            now
          </span>
          <div
            class="d-flex"
          >
            <span>
              Users
              :
            </span>
            <div
              class="ms-auto d-flex gap-1"
            >
              <div
                class="d-flex board-users-avatar-sm bg-secondary text-light border rounded-circle align-items-center justify-content-center"
              >
                U
              </div>
              <div
                class="d-flex board-users-avatar-sm bg-secondary text-light border rounded-circle align-items-center justify-content-center"
              >
                U
              </div>
            </div>
          </div>
        </div>
        <hr
          class="mb-2 mt-2"
        />
        <button
          class="btn btn-primary border rounded-2"
          data-bs-target="#addTask"
          data-bs-toggle="modal"
          data-testid="show-add-task"
        >
          <svg
            class="bi bi-plus"
            fill="currentColor"
            height="24"
            viewBox="0 0 16 16"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 4a.5.5 0 0 1 .5.5v3h3a.5.5 0 0 1 0 1h-3v3a.5.5 0 0 1-1 0v-3h-3a.5.5 0 0 1 0-1h3v-3A.5.5 0 0 1 8 4"
            />
          </svg>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="rounded-2 d-flex flex-column"
    >
      <div
        class="d-flex flex-column board-task cursor-pointer cursor-grab"
        data-bs-target="#changeTask"
        data-bs-toggle="modal"
        data-testid="task1"
        draggable="true"
      >
        <span
          class="fs-5"
        >
          Task 1
        </span>
        <span>
          Moved
          :
        </span>
        <span
          class="text-truncate"
        >
          54 years ago
        </span>
        <span>
          Created
          :
        </span>
        <span
          class="text-truncate"
        >
          54 years ago
        </span>
        <div
          class="d-flex"
        >
          <span>
            Users
            :
          </span>
          <div
            class="ms-auto d-flex gap-1"
          >
            <div
              class="d-flex board-users-avatar-sm bg-secondary text-light border rounded-circle align-items-center justify-content-center"
            >
              U
            </div>
            <div
              class="d-flex board-users-avatar-sm bg-secondary text-light border rounded-circle align-items-center justify-content-center"
            >
              U
            </div>
            <div
              class="d-flex board-users-avatar-sm bg-secondary text-light border rounded-circle align-items-center justify-content-center"
            >
              U
            </div>
          </div>
        </div>
      </div>
      <div
        class="d-flex flex-column board-task cursor-pointer cursor-grab"
        data-bs-target="#changeTask"
        data-bs-toggle="modal"
        data-testid="task2"
        draggable="true"
      >
        <span
          class="fs-5"
        >
          Task 2
        </span>
        <span>
          Moved
          :
        </span>
        <span
          class="text-truncate"
        >
          now
        </span>
        <span>
          Created
          :
        </span>
        <span
          class="text-truncate"
        >
          now
        </span>
        <div
          class="d-flex"
        >
          <span>
            Users
            :
          </span>
          <div
            class="ms-auto d-flex gap-1"
          >
            <div
              class="d-flex board-users-avatar-sm bg-secondary text-light border rounded-circle align-items-center justify-content-center"
            >
              U
            </div>
            <div
              class="d-flex board-users-avatar-sm bg-secondary text-light border rounded-circle align-items-center justify-content-center"
            >
              U
            </div>
          </div>
        </div>
      </div>
      <hr
        class="mb-2 mt-2"
      />
      <button
        class="btn btn-primary border rounded-2"
        data-bs-target="#addTask"
        data-bs-toggle="modal"
        data-testid="show-add-task"
      >
        <svg
          class="bi bi-plus"
          fill="currentColor"
          height="24"
          viewBox="0 0 16 16"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 4a.5.5 0 0 1 .5.5v3h3a.5.5 0 0 1 0 1h-3v3a.5.5 0 0 1-1 0v-3h-3a.5.5 0 0 1 0-1h3v-3A.5.5 0 0 1 8 4"
          />
        </svg>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Tasks should render only the add task button when column has no tasks 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rounded-2 d-flex flex-column"
      >
        <button
          class="btn btn-primary border rounded-2"
          data-bs-target="#addTask"
          data-bs-toggle="modal"
          data-testid="show-add-task"
        >
          <svg
            class="bi bi-plus"
            fill="currentColor"
            height="24"
            viewBox="0 0 16 16"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 4a.5.5 0 0 1 .5.5v3h3a.5.5 0 0 1 0 1h-3v3a.5.5 0 0 1-1 0v-3h-3a.5.5 0 0 1 0-1h3v-3A.5.5 0 0 1 8 4"
            />
          </svg>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="rounded-2 d-flex flex-column"
    >
      <button
        class="btn btn-primary border rounded-2"
        data-bs-target="#addTask"
        data-bs-toggle="modal"
        data-testid="show-add-task"
      >
        <svg
          class="bi bi-plus"
          fill="currentColor"
          height="24"
          viewBox="0 0 16 16"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 4a.5.5 0 0 1 .5.5v3h3a.5.5 0 0 1 0 1h-3v3a.5.5 0 0 1-1 0v-3h-3a.5.5 0 0 1 0-1h3v-3A.5.5 0 0 1 8 4"
          />
        </svg>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
